scripts/masterCron.ts

```diff
@@ -0,0 +1,272 @@
+  await sleep(auctionDetails.auctionEndDate.sub(await time.now()).mul(1000));
+}
+
+async function settleAuctions(
+  gnosisAuction: Contract,
+  provider: any,
+  signer: Wallet,
+  network: string,
+  auctionCounters: Array<number>
+) {
+  for (let auctionID in auctionCounters) {
+    const auctionDetails = await gnosisAuction.auctionData(auctionID);
+    // If initialAuctionOrder is bytes32(0) auction has
+    // already been settled as gnosis does gas refunds
+    if (auctionDetails.initialAuctionOrder === BYTES_ZERO) {
+      continue;
+    }
+    let gasPrice = await gas(network);
+    try {
+      const tx = await gnosisAuction.connect(signer).settleAuction({
+        gasPrice,
+      });
+      await log(`settleAuction (${auctionID}): ${tx.hash}`);
+    } catch (error) {
+      await log(
+        `@everyone settleAuction (${auctionID}): failed with error ${error}`
+      );
+    }
+  }
+}
+
+async function runTX(
+  vaultArtifactAbi: any,
+  provider: any,
+  signer: Wallet,
+  network: string,
+  method: string
+) {
+  let returnData = [];
+  for (let vaultName in deployments[network].vaults) {
+    const vault = new ethers.Contract(
+      deployments[network].vaults[vaultName],
+      vaultArtifactAbi,
+      provider
+    );
+
+    // If current option is not zero address, means
+    // someone already called new weeks rollToNextOption
+    if (
+      method === "rollToNextOption" &&
+      (await vault.currentOption()) === constants.AddressZero
+    ) {
+      await log(`${method} (${vaultName}): skipped`);
+      continue;
+    }
+
+    let gasPrice = await gas(network);
+
+    try {
+      const tx = await vault.connect(signer)[`${method}()`]({
+        gasPrice,
+      });
+      log(`${method} (${vaultName}): ${tx.hash}`);
+
+      if (method === "commitAndClose") {
+        returnData[0] = (await vault.delay()).mul(1000);
+      } else if (method === "rollToNextOption") {
+        const receipt = await tx.wait();
+        returnData.push(
+          hexStripZeros(receipt["logs"][15]["topics"][1]).toString().slice(2)
+        );
+      }
+    } catch (error) {
+      await log(
+        `@everyone ${method} (${vaultName}): failed with error ${error}`
+      );
+    }
+  }
+
+  return returnData;
+}
+
+async function masterJob() {
+  const vaultArtifact = await hre.artifacts.readArtifact("RibbonThetaVault");
+  const gnosisArtifact = await hre.artifacts.readArtifact("IGnosisAuction");
+
+  const gnosisAuction = new ethers.Contract(
+    GNOSIS_EASY_AUCTION,
+    gnosisArtifact.abi,
+    provider
+  );
+
+  // Master Cron Job
+  //
+  // 1. commitAndClose
+  let delayBeforeRoll = await runTX(
+    vaultArtifact.abi,
+    provider,
+    signer,
+    network,
+    "commitAndClose"
+  );
+  // 2. wait an hour (timelock period)
+  await sleep(delayBeforeRoll[0]);
+  // 3. rollToNextOption
+  let auctionCounters = await runTX(
+    vaultArtifact.abi,
+    provider,
+    signer,
+    network,
+    "rollToNextOption"
+  );
```

Comment:
What if we just created a few cron jobs instead of sleeping for an hour? Better to decouple them imo